shimmingtoolbox/prepare_fieldmap.py

```diff
@@ -101,3 +116,46 @@ def prepare_fieldmap(list_nii_phase, echo_times, unwrapper='prelude', mag=None,
 
     # return fieldmap_hz_gaussian
     return fieldmap_hz
+
+
+def correct_2pi_offset(unwrapped, mag, mask, validity_threshold):
+    """ Removes 2*pi offsets from `unwrapped` for a time series. If there is no offset, it returns the same array.
+
+    Args:
+        unwrapped (numpy.ndarray): 4d array of the spatially unwrapped phase
+        mag (numpy.ndarray): 4d array containing the magnitude values of the phase
+        mask (numpy.ndarray): 4d mask of the unwrapped phase array
+        validity_threshold (float): Threshold to create a mask on each timepoints and assume as reliable phase data
+
+    Returns:
+        numpy.ndarray: 4d array of the unwrapped phase corrected if there were n*2*pi offsets between time points
+
+    """
+    # Create a mask that excludes the noise
+    validity_masks = mask_threshold(mag - mag.min(), validity_threshold * (mag.max() - mag.min()))
+
+    for i_time in range(1, unwrapped.shape[3]):
+        # Take the region where both masks intersect
+        validity_mask = np.logical_and(validity_masks[..., i_time - 1], validity_masks[..., i_time])
+
+        # Calculate the means in the same validity mask
+        ma_0 = np.ma.array(unwrapped[..., i_time - 1], mask=validity_mask == False)
+        mean_0 = np.ma.mean(ma_0)
+        ma_1 = np.ma.array(unwrapped[..., i_time], mask=validity_mask == False)
+        mean_1 = np.ma.mean(ma_1)
+
+        # Calculate the number of offset by rounding to the nearest integer.
+        n_offsets_float = (mean_1 - mean_0) / (2 * np.pi)
+        n_offsets = round(n_offsets_float)
```

Comment:
i would suggest not rounding, and testing explicitly in the code if value is between -0.2 and 0.2 (or whatever threshold you choose)